<launch>

  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find navlab_turtlebot_sim)/maps/flightroom.yaml"/>
  <arg name="open_rviz" default="true"/>

  <!-- Flightroom launch -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find navlab_turtlebot_sim)/worlds/flightroom_obstacles.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- Goal Planner -->
  <node pkg="navlab_turtlebot_sim" name="goal_planner" type="goal_planner.py" output="screen" />

  <!-- BEGIN ROBOT 1-->
  <group ns="turtlebot1">
    <param name="tf_prefix" value="turtlebot1_tf" />
    <include file="$(find navlab_turtlebot_sim)/launch/sim_one_robot.launch" >
      <arg name="x_pos" value="2" />
      <arg name="y_pos" value="2" />
      <arg name="yaw" value="0" />
      <arg name="robot_name"  value="turtlebot1" />
    </include>
  </group>

  <!-- BEGIN ROBOT 2-->
  <group ns="turtlebot2">
    <param name="tf_prefix" value="turtlebot2_tf" />
    <include file="$(find navlab_turtlebot_sim)/launch/sim_one_robot.launch" >
      <arg name="x_pos" value="2" />
      <arg name="y_pos" value="-2" />
      <arg name="yaw" value="4.71" />
      <arg name="robot_name"  value="turtlebot2" />
    </include>
  </group>

  <!-- BEGIN ROBOT 3-->
  <group ns="turtlebot3">
    <param name="tf_prefix" value="turtlebot3_tf" />
    <include file="$(find navlab_turtlebot_sim)/launch/sim_one_robot.launch" >
      <arg name="x_pos" value="-2" />
      <arg name="y_pos" value="-2" />
      <arg name="yaw" value="3.14" />
      <arg name="robot_name"  value="turtlebot3" />
    </include>
  </group>

  <!-- BEGIN ROBOT 4-->
  <group ns="turtlebot4">
    <param name="tf_prefix" value="turtlebot4_tf" />
    <include file="$(find navlab_turtlebot_sim)/launch/sim_one_robot.launch" >
      <arg name="x_pos" value="-2" />
      <arg name="y_pos" value="2" />
      <arg name="yaw" value="1.57" />
      <arg name="robot_name"  value="turtlebot4" />
    </include>
  </group>

  <!-- rviz -->
  <group if="$(arg open_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find navlab_turtlebot_sim)/rviz/turtlebot_four_navigation.rviz"/>
  </group>

</launch>
